main.cpp

```diff
@@ -273,6 +383,28 @@ struct Fingerboard
     bool requireNewFrets();//returns true if the fretboard requires new frets
 
 };
+
+int Fingerboard::fretANote(int stringNumber, int fretNumber)
+{
+    return stringNumber + fretNumber + 9000;
+}
+std::string Fingerboard::requireOil(bool required)
+{
+    if(required)
+    {
+        return "yes";
+    }
+    else
+    {
+        return "no";
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise